src/handlers/expiry/providers/TestDataProvider.ts

```diff
@@ -22,61 +17,66 @@ export class TestDataProvider implements ITestDataProvider {
+          .filter((test) => test.testStatus === enums.TEST_STATUS.SUBMITTED)
           .filter(
             (testResult: {
               testStartTimestamp: string | number | Date;
               testEndTimestamp: string | number | Date;
             }) => {
-              return (
-                moment(testResult.testStartTimestamp).isAfter(fromDateTime) &&
-                moment(testResult.testEndTimestamp).isBefore(toDateTime)
+              const { testStartTimestamp, testEndTimestamp } = testResult;
+              if (!DateProvider.isValidDate(testStartTimestamp) || !DateProvider.isValidDate(testEndTimestamp)) {
+                throw new Error(`Invalid timestamp -> StartTimestamp: ${testStartTimestamp} EndTimeStamp: ${testEndTimestamp} `);
+              }
+              return DateProvider.isBetweenDates(
+                testStartTimestamp,
+                testEndTimestamp,
+                fromDateTime,
+                toDateTime
               );
             }
           );
       }
       return result;
     } catch (error) {
-      if (!(error instanceof HTTPError)) {
-        console.log(error);
-        error = new HTTPError(500, MESSAGES.INTERNAL_SERVER_ERROR);
-      }
+      console.error("getTestHistory: Error ", error);
+      error = new HTTPError(500, enums.MESSAGES.INTERNAL_SERVER_ERROR);
       throw error;
     }
   }
 
   public async getMostRecentExpiryDate(systemNumber: string): Promise<Date> {
     let maxDate = DateProvider.getEpoc();
     const testResults = await this.getTestHistory(systemNumber);
-    if (testResults && testResults.length) {
-      const filteredTestTypeDates: any[] = [];
-      testResults.forEach(({ testTypes }) => {
+    console.log( `getMostRecentExpiryDate: Filtered Data Count -> ${testResults?.length}`);
+
+    const filteredTestTypeDates: any[] = [];
+    testResults.forEach(({ testTypes }) => {
         testTypes.forEach(({ testCode, testExpiryDate }) => {
-          if (
-            testCode &&
-            TestDataProvider.isValidTestCodeForExpiryCalculation(
-              testCode.toUpperCase()
-            ) &&
-            testExpiryDate &&
-            DateProvider.isValidDate(testExpiryDate)
-          ) {
-            filteredTestTypeDates.push(moment(testExpiryDate));
+
+          if (testCode && TestDataProvider.isValidTestCodeForExpiryCalculation(testCode.toUpperCase()) && testExpiryDate) {
+            if (!DateProvider.isValidDate(testExpiryDate)) {
+                throw new Error(`Invalid Expiry Date: ${testExpiryDate}`);
+              }
+            console.log(`getMostRecentExpiryDate: Filtered Date -> ${testExpiryDate}`);
+            filteredTestTypeDates.push(DateProvider.getInstance((testExpiryDate)));
           }
-        });
+
       });
-      if (filteredTestTypeDates.length) {
-        maxDate = moment.max(filteredTestTypeDates).toDate();
+        if (filteredTestTypeDates.length) {
```

Comment:
I think we need to add prettier at some point

Reply:
Will be handled in CVSB-18734